### PR TITLE
Add cache busting header & improve image validation time when import

### PR DESCRIPTION
**Pull request by YCCertifai**
# Description

### Cache Busting
Same logic as #312

### Image Validation Time Improvement
Same logic as #339 with changes made as per request by @codenamewei 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Tested on?

- [ ] Windows  
- [x] Linux Ubuntu 
- [ ] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Review by codenamewei — CHANGES_REQUESTED** [review]
removal of classifai-distro folder..

File: classifai-core/src/main/java/ai/classifai/data/type/image/JpegImageData.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class JpegImageData extends ImageData
+{
+    public JpegImageData(Metadata metadata)
+    {
+        super(metadata);
+    }
+
+    @Override
+    public int getWidth() throws MetadataException {
+        return metadata.getFirstDirectoryOfType(JpegDirectory.class).getImageWidth();
```
Comment (codenamewei) [inline classifai-core/src/main/java/ai/classifai/data/type/image/JpegImageData.java:36]: any reason why jpeg is not doing 
`metadata.getFirstDirectoryOfType(JpegDirectory.class).getInt(JpegDirectory.TAG_IMAGE_WIDTH);`

which conforms with the rest (bmp, png)?
@YCCertifai

**Review by YCCertifai — COMMENTED** [review]
> removal of classifai-distro folder..

fixed

**Comment by YCCertifai** [comment]
changes made from the last commit:
1. Merged with the latest v2_alpha branch
2. JpegImageData getWidth() and getHeight() methods style changed to conform with others ImageData
3. Moved Exceptions class to util/exception
4. Refactor add "no-cache" into a helper method to keep code clean

@codenamewei 
kindly review it whenever you are free. Thanks!

**Review by codenamewei — CHANGES_REQUESTED** [review]
@YCCertifai, 
Tried a folder of 564 images on Mac and faced errors as below
https://gist.github.com/codenamewei/efa168868269c51f399c4496fe65f2d0

**Comment by YCCertifai** [comment]
> @YCCertifai,
> Tried a folder of 564 images on Mac and faced errors as below
> https://gist.github.com/codenamewei/efa168868269c51f399c4496fe65f2d0

I've tested on 1k images and also some "fake image"(random file with ".jpg" extension)
still do not fall into this exception...
the link to the images is [here](https://drive.google.com/drive/folders/1fc6BfSILAgl2OXT_FZvlq_Nf72tAzMBz?usp=sharing)

If possible please provide the images you have tested on, so I could debug it...
thanks

**Comment by codenamewei** [comment]
@YCCertifai 
Did a brief look into 
Local path of image: _/Users/chiawei/Desktop/sample-data/Education/stationery/ruler.jpg_
Path of image in application: _/Users/chiawei/Desktop/sample-data/Education/ruler.jpg_

Dataset Folder: [access by company email](https://bit.ly/334Fe9p)

How to reproduce:
Import **Education** folder or any folder which contains subfolder of images

Environment: tried on Mac & Windows

**Comment by YCCertifai** [comment]
bug fixed.

Tested with cases:
1. import folder with subfolders
2. import/export 
3. reload
3. import/export with data (including cross-platform project importing)

**Review by devenyantis — APPROVED** [review]

**Review by codenamewei — APPROVED** [review]

**Pull request merged**